src/tiles.rs

```diff
@@ -0,0 +1,401 @@
+                    let new_tile_id = self.insert_tile(tile);
+                    let mut tabs = Tabs::new(vec![new_tile_id]);
+                    tabs.children.insert(index.min(1), child_id);
+                    tabs.active = child_id;
+                    self.tiles
+                        .insert(parent_id, Tile::Container(Container::Tabs(tabs)));
+                }
+            }
+            LayoutInsertion::Horizontal(index) => {
+                if let Tile::Container(Container::Linear(Linear {
+                    dir: LinearDir::Horizontal,
+                    children,
+                    ..
+                })) = &mut tile
+                {
+                    let index = index.min(children.len());
+                    children.insert(index, child_id);
+                    self.tiles.insert(parent_id, tile);
+                } else {
+                    let new_tile_id = self.insert_tile(tile);
+                    let mut linear = Linear::new(LinearDir::Horizontal, vec![new_tile_id]);
+                    linear.children.insert(index.min(1), child_id);
+                    self.tiles
+                        .insert(parent_id, Tile::Container(Container::Linear(linear)));
+                }
+            }
+            LayoutInsertion::Vertical(index) => {
+                if let Tile::Container(Container::Linear(Linear {
+                    dir: LinearDir::Vertical,
+                    children,
+                    ..
+                })) = &mut tile
+                {
+                    let index = index.min(children.len());
+                    children.insert(index, child_id);
+                    self.tiles.insert(parent_id, tile);
+                } else {
+                    let new_tile_id = self.insert_tile(tile);
+                    let mut linear = Linear::new(LinearDir::Vertical, vec![new_tile_id]);
+                    linear.children.insert(index.min(1), child_id);
+                    self.tiles
+                        .insert(parent_id, Tile::Container(Container::Linear(linear)));
+                }
+            }
+            LayoutInsertion::Grid(insert_location) => {
+                if let Tile::Container(Container::Grid(grid)) = &mut tile {
+                    grid.locations.retain(|_, pos| *pos != insert_location);
+                    grid.locations.insert(child_id, insert_location);
+                    grid.children.push(child_id);
+                    self.tiles.insert(parent_id, tile);
+                } else {
+                    let new_tile_id = self.insert_tile(tile);
+                    let mut grid = Grid::new(vec![new_tile_id, child_id]);
+                    grid.locations.insert(child_id, insert_location);
+                    self.tiles
+                        .insert(parent_id, Tile::Container(Container::Grid(grid)));
+                }
+            }
+        }
+    }
+
+    pub(super) fn gc_root(&mut self, behavior: &mut dyn Behavior<Pane>, root_id: TileId) {
```

Comment:
for a lil bit I was struggling with the distinction between gc & simplify. I think it would be good to call out here that gc is all about removing unconnected/unknown/remove tiles whereas simplify deals with actual tree structure in an opinionated (and configurable) way